### PR TITLE
network: Add redis infrastructure

### PR DESCRIPTION
**Pull request by netoax**
**Describe what this PR introduces:**

This PR adds the basic infrastructure to communicate with the Redis server.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix
- [X] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce any breaking changes?** (check one)

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] Related issues are referenced in the PR (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [X] All tests are passing
- [ ] New/updated tests are included

**Testing environment:**

- Operating System/Platform: MacOS - Darwin 18.0.0
- Go version: 1.16

If you have relevant logs, error output and etc, please attach them.

**Other information:**

### CODE REVIEW

**Review by lcbm — APPROVED** [review]
LGTM! :smile:

File: pkg/network/redis.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+// of any supported type: https://redis.io/topics/data-types.
+func (r *Redis) Set(key string, value interface{}, expiration time.Duration) error {
+	return r.rdb.Set(ctx, key, value, expiration).Err()
+}
+
+// Get retrieves a value from the Redis database according to key, which is returned as a string.
+func (r *Redis) Get(key string) (string, error) {
+	val, err := r.rdb.Get(ctx, key).Result()
+	if err != nil {
+		return "", nil
```
Comment (vitbaq) [inline pkg/network/redis.go]: Should you return err here?

File: cmd/main.go [hunk 2]
```diff
@@ -39,6 +39,10 @@ func main() {
 
 	go monitorSignals(sigs, quit, logger)
 
+	// Redis
+	redis := network.NewRedis(config.Redis.URL, logrus.Get("Redis"))
+	redisStartedChan := make(chan bool, 1)
```
Comment (vitbaq) [inline cmd/main.go]: Just do this if you have another thing to change.
- Change the order just to keep the pattern of the code.

You can close this thread if necessary.

File: go.mod [hunk 3]
```diff
@@ -5,7 +5,6 @@ go 1.16
 require (
```
Comment (vitbaq) [inline go.mod:5]: what is the reason for the commit 40b7a0c24eff6c8a9acb65af181914428fcfec97 change?
should it come before the previous commit?

**Review by vitbaq — CHANGES_REQUESTED** [review]

File: pkg/network/redis.go (hunk 1 above)
Comment (netoax) [reply pkg/network/redis.go]: Done.

File: cmd/main.go (hunk 2 above)
Comment (netoax) [reply cmd/main.go]: Done.

File: go.mod (hunk 3 above)
Comment (netoax) [reply go.mod:5]: Done.

**Review by vitbaq — APPROVED** [review]

**Pull request merged**